lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -148,14 +168,17 @@ defmodule KinesisClient.Stream.Shard.Producer do
         "shard_id: #{state.shard_id}"
     )
 
-    state = handle_closed_shard(state)
-
-    {:noreply, [], state}
+    if state.status == :closed do
```

Comment:
I think I need to restore this bit, now that `status == :closed` is being set inside of handle_closed_shard.